homeassistant/components/velbus/__init__.py

```diff
@@ -138,9 +139,17 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
     await hass.data[DOMAIN][entry.entry_id]["cntrl"].stop()
     hass.data[DOMAIN].pop(entry.entry_id)
+    await hass.async_add_executor_job(
+        shutil.rmtree,
```

Comment:
This will require a complete rewrite of the velbus object caching.
Now this just thing just expects a directory where it can store the files per address.

Reply:
Yeah I completely understand, but this is a bugfix for a user, so would really like this to be merged and I will start working on the storage helper implementation.

Althought I'm not 100% how this would work out.